server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2032,7 +2037,8 @@ private static InputStream maybeRateLimit(InputStream stream, @Nullable RateLimi
     }
 
     public InputStream maybeRateLimitRestores(InputStream stream) {
-        return maybeRateLimit(stream, restoreRateLimiter, restoreRateLimitingTimeInNanos);
+        return maybeRateLimit(maybeRateLimit(stream, restoreRateLimiter, restoreRateLimitingTimeInNanos),
+            recoverySettings.rateLimiter(), restoreRateLimitingTimeInNanos);
```

Comment:
`recoverySettings.rateLimiter()` doesn't stay constant if you completely enable/disable recovery rate limiting. In peer recovery we get it anew for each file chunk sent/received, whereas here a change may only take effect at the start of a blob.

Reply:
I've pushed a576d8d